designs/drift-hash-versioning.md

```diff
@@ -0,0 +1,155 @@
+# Karpenter Drift Hash Versioning with CRD
+
+## Summary of Drift Hashing Mechanism
+
+Drift utilizes a hashing mechanism to identify when a `NodeClaim` has been drifted from its owning `NodePool`. Karpenter will hash fields that are within the `NodePool.Spec.Template`, excluding [special cases of drift](https://karpenter.sh/docs/concepts/disruption/#special-cases-on-drift) or [behavior fields](https://karpenter.sh/docs/concepts/disruption/#behavioral-fields). The `NodePool` will be annotated with the hash. When a `NodeClaim` is created, the `NodePool` hash will be annotated on the `NodeClaim`. More details can be found in the [drift documentation](https://karpenter.sh/docs/concepts/disruption/#drift). 
+
+When a `NodePool` is updated, a new hash will be annotated on the `NodePool`. The `NodeClaim` disruption controller will periodically check if the `NodePool` hash match to determine when a `NodeClaim` is drifted.
+
+## Background
+
+Karpenter's static drift mechanism works well when adding new fields to the CRD that don't have defaults, but breaks down if we add a field that has a set default, remove a field from the CRD, or set a new default for an existing field. If a cloud provider makes one of these changes to their CRDs, the static drift would change, causing all nodes to roll when the CRD is upgraded. This causes problems for cloud providers when maintainers need to make these kinds of changes. We currently have three issue that need a drift hash versioning improvement: 
```

Comment:
I have a recommendation up above to call that section "Summary of Static Drift Hashing Mechanism" so that when you come down here to talk about "Karpenter's static drift mechanism" it's immediately clear what you are referring to. That gives you some nice parallelism in your writing